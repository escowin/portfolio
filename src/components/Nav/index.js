import React from "react";

function Nav() {
    const catagories = [
        {
            name: "Solo work",
            description: "solo projects",
        },
        {
            name: "Collaborations",
            description: "collaborative apps"
        }
    ];

    function categorySelected(name) {
        console.log(`${name} clicked`);
    }

    return (
        <>
            <header id="header">
                <h1>Edwin m. escobar</h1>
                <p><a data-testid="link" href="/">The react portfolio</a> <span role="img" aria-label="camera"> ðŸ“¸</span> </p>
            </header>
            <nav id="navigation">
                <ul>
                    <li className="nav-link">
                        <a data-testid="about" href="#about">About</a>
                    </li>
                    <li className="nav-link">
                        <span>Contact</span>
                    </li>
                    {catagories.map((category) => (
                        <li className="nav-link" key={category.name}>
                            <span onClick={() => categorySelected(category.name)} >
                                {category.name}
                            </span>
                        </li>
                    ))}
                    <li className="nav-link">
                        <a href=".">Resume</a>
                    </li>
                </ul>
            </nav>
        </>
    );
}

export default Nav;