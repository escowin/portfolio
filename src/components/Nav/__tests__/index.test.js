import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import Nav from "..";

describe ('Nav component', () => {
    // baseline test
    it('renders', () => {
        render(<Nav/>);
    });

    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav/>);
        // assert value comparison
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('emoji is visible', () => {
    it('inserts camera emoji into p', () => {
        // arrange
        const { getByLabelText } = render(<Nav/>);
        // assert
        expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
    })
})

describe('links are visible', () => {
    it('inserts text into the links', () => {
        // arrange query
        const { getByTestId } = render(<Nav/>);
        // assert
        expect(getByTestId('link')).toHaveTextContent('The react portfolio');
        expect(getByTestId('about')).toHaveTextContent('About');
    });
})

afterEach(cleanup);