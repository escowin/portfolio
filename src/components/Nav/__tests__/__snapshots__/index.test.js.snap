// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header
    id="header"
  >
    <h1>
      Edwin m. escobar
    </h1>
    <p>
      <a
        data-testid="link"
        href="/"
      >
        The react portfolio
      </a>
       
      <span
        aria-label="camera"
        role="img"
      >
         ðŸ“¸
      </span>
       
    </p>
  </header>
  <nav
    id="navigation"
  >
    <ul>
      <li
        class="nav-link"
      >
        <a
          data-testid="about"
          href="#about"
        >
          About
        </a>
      </li>
      <li
        class="nav-link"
      >
        <span>
          Contact
        </span>
      </li>
      <li
        class="nav-link"
      >
        <span>
          Solo work
        </span>
      </li>
      <li
        class="nav-link"
      >
        <span>
          Collaborations
        </span>
      </li>
      <li
        class="nav-link"
      >
        <a
          href="."
        >
          Resume
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
